client/src/api/v2/camera.ts

```diff
@@ -0,0 +1,298 @@
+  ];
+
+  // Format data that is dependent on status
+  const formatter = CameraRecordingStatusPayload.match(
+    () => [],
+    (onPayload) => [
+      {
+        name: 'Recording duration',
+        value: formatMinutes(onPayload.recordingMinutes),
+      },
+      { name: 'Recording file', value: onPayload.recordingFile },
+    ],
+    (errorPayload) => [{ name: 'Error', value: errorPayload.error }],
+  );
+
+  return commonData.concat(formatter(payload));
+}
+
+/**
+ * Initiate receiving status payloads
+ *
+ * @param subComponent The sub component to init for (e.g. recording, video feed)
+ * @param device Device
+ */
+function initStatus(subComponent: string, device?: CameraDevice) {
+  const path = ['camera', subComponent];
+  if (device) path.push(device);
+  emit(`get-status-payload`, path);
+}
+
+/**
+ * Makes the device name pretty.
+ *
+ * Hardcoded for efficiency. If you don't like it, complain to Angus Trau :).
+ *
+ * @param device Device
+ * @returns  Prettied device name
+ */
+export function getPrettyDeviceName(device: CameraDevice) {
+  return device === 'primary' ? 'Primary' : 'Secondary';
+}
+
+/**
+ * `T` Type of the socket.io payload
+ * `U` Type to be returned by the handler
+ */
+interface StatusPayloadOptions<T, U> {
+  /** Initial value for the payload */
+  initValue: T | null;
+  /** Handler for updating payload */
+  payloadHandler?: (
+    setter: React.Dispatch<React.SetStateAction<T | null>>,
+    newPayload: T | null,
+    device: CameraDevice,
+  ) => T | void;
+  /** Handler for return value */
+  returnHandler?: (payload: T | null, device?: CameraDevice) => U | null;
+}
+
+/**
+ * Create hooks for getting payloads
+ *
+ * @param sub Subcomponent
+ * @param shape Shape of received payloads
+ * @param opts Options
+ * @returns Hook for getting a payload
+ */
+function createStatusPayloadHook<T>(
+  sub: string,
+  shape: Runtype<T>,
+  {
+    initValue = null,
+    payloadHandler = (
+      setter: React.Dispatch<React.SetStateAction<T | null>>,
+      newPayload: T | null,
+    ) => setter(newPayload ?? initValue),
+    returnHandler = (payload: T | null) => payload,
+  }: StatusPayloadOptions<T, T>,
+) {
+  return function _hook(device: CameraDevice) {
+    // Only run init once per render
+    useEffect(() => {
+      initStatus(sub, device);
+    }, [device]);
+
+    const [payload, setPayload] = useState(initValue);
+    useChannelShaped(`status-camera-${sub}`, shape, (newPayload: T) =>
+      payloadHandler(setPayload, newPayload, device),
+    );
+
+    return returnHandler(payload, device);
+  };
+}
+
+const CameraRecordingStatusPayloads = Partial({
+  primary: CameraRecordingStatusPayload,
+  secondary: CameraRecordingStatusPayload,
+});
+
+/**
+ * Returns the last status payload published
+ *
+ * @param device Device
+ * @returns Recording status
+ */
+export const useCameraRecordingStatus = createStatusPayloadHook(
+  'recording',
+  CameraRecordingStatusPayloads,
+  {
+    initValue: null,
+  },
```

Comment:
Could you remind me why `returnHandler` was removed from this and why `parseRecordingPayloadData` was also removed? [See here](https://github.com/monash-human-power/dashboard/blob/master/client/src/api/v2/camera.js#L204)

Reply:
Not sure if it was the original reason for moving it or not but I'm just too little brained to get the logic that had been moved to `CameraRecordingStatus` back into a `returnHandler`. If it's important, would you be able to lend a hand with how to do this and make typescript happy?

`parseRecordingPayloadData` hasn't been removed, only renamed to `formatRecordingPayload`.

Reply:
Well if it runs with the TS `CameraRecordingStatus` then it should be fine. I'll check it out when this PR gets merged.